portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -65,10 +74,31 @@ static Comment fromEntity(Entity e) {
         return new Comment(
           (String) e.getProperty("text"),
           (String) e.getProperty("name"),
-          (long) e.getProperty("time"));
+          (long) e.getProperty("time"), 
+          (String) e.getProperty("id"));
       }
   }
 
+  /** 
+    * Iterates over all persistent comments upon server startup
+    * and assigns each a unique id number, then decides the value of
+    * the next valid unique id number that can be assigned to a 
+    * future comment. 
+    */
+  @Override
+  public void init() { 
```

Comment:
This approach comes with some long term downsides because every server startup will modify All Entities. 

This is particularly problematic when we consider that all modern websites today are served with multiple servers. Therefore, in practice, this approach would cause concurrent access issues and heavy database usage. 

Instead, consider using a different ID mechanism - it's possible that the database already provides you with Unique Ids (what about Keys?).

Reply:
Ohhhh I should be using the keys!! I can't believe I didn't think of that; I'll go implement that!

Reply:
Okay, using key-ids as ids is implemented!